firebase-appdistribution-api/ktx/src/androidTest/kotlin/com/google/firebase/app/distribution/ktx/FirebaseAppDistributionTests.kt

```diff
@@ -70,6 +74,43 @@ class FirebaseAppDistributionTests : BaseTestCase() {
     fun appDistribution_default_callsDefaultGetInstance() {
         assertThat(Firebase.appDistribution).isSameInstanceAs(FirebaseAppDistribution.getInstance())
     }
+
+    @Test
+    fun appDistributionReleaseDestructuringDeclarationsWork() {
+        val mockAppDistributionRelease = object : AppDistributionRelease {
+            override fun getDisplayVersion(): String = "1.0.0"
+
+            override fun getVersionCode(): Long = 1L
+
+            override fun getReleaseNotes(): String = "Changelog..."
+
+            override fun getBinaryType(): BinaryType = BinaryType.AAB
+        }
+
+        val (type, displayVersion, versionCode, notes) = mockAppDistributionRelease
+
+        assertThat(type).isSameInstanceAs(mockAppDistributionRelease.binaryType)
```

Comment:
I think you should use `isEqualTo()` instead of `isSameInstanceAs()`.

For singletons like enums those are identical, but (boxed) Longs (or Ints) are no guaranteed to re-use the same instance for the same values.